src/shims/unix/fd.rs

```diff
@@ -31,6 +31,7 @@ pub trait FileDescription: std::fmt::Debug + Any {
         _self_ref: &FileDescriptionRef,
         _communicate_allowed: bool,
         _bytes: &mut [u8],
+        _ptr: Pointer,
```

Comment:
Please add a doc comment explaining what the pointer is for.

Why is this not `dest: &MPlaceTy<'tcx>`?

Reply:
The pointer is the buf pointer provided by the caller, it is used in ``read_byte_helper`` to write to user supplied buffer.

Reply:
This might be able optimised further, but how ``read`` currently works is:
1. In ``FileDescription::read``, the bytes is written into ``&mut [u8]``
2. ``&mut [u8]`` is passed into ``read_byte_helper``
3. In ``read_byte_helper``, the data is copied from ``&mut [u8]`` to the user supplied buffer. 

I am not sure if there is a convenient way to move step 1 into ``read_byte_helper``, because read for stdin uses ``Read::read(&mut { self }, bytes)`` where eventfd uses 
```rust            
*bytes = match ecx.tcx.sess.target.endian {
                Endian::Little => counter.to_le_bytes(),
                Endian::Big => counter.to_be_bytes(),
};
```

Reply:
We should definitely not have both the `bytes` buffer and the `ptr` here, that's a too confusing API.

Reply:
So replace `bytes` by `len: usize`, and in `eventd` you should deref the `buf: Pointer` as type `u32` -- that's a lot nicer than what we do right now, in fact.